packages/amplify-category-geo/src/service-utils/resourceUtils.ts

```diff
@@ -190,17 +190,63 @@ export const getServicePermissionPolicies = (
 }
 
 export const getAuthResourceName = async (context: $TSContext): Promise<string> => {
-  let authResources = (await context.amplify.getResourceStatus(authCategoryName)).allResources;
-  authResources = authResources.filter((resource: $TSObject) => resource.service === 'Cognito');
-
-  if (authResources.length === 0) {
-    throw new Error('No auth resource found. Please add it using "amplify add auth"');
+  const authMeta = stateManager.getMeta()?.[authCategoryName];
+  const cognitoResources = authMeta ? Object.keys(authMeta).filter(authResource => authMeta[authResource].service === 'Cognito') : [];
+  if (cognitoResources.length === 0) {
+    throw new Error('No auth resource found. Run "amplify add auth"');
   }
-  return authResources[0].resourceName;
+  return cognitoResources[0];
 }
 
 export type ResourceDependsOn = {
   category: string;
   resourceName: string;
   attributes: string[];
 };
+
+/**
+ * Construct the resource dependencies on other category resources
+ * @param groupNames Cognito groups that are granted permissions for the resource
+ * @param authResourceName Name of the auth category resource added
+ */
+export const getResourceDependencies = (groupNames: string[], authResourceName: string): ResourceDependsOn[] => {
+  const dependsOnResources = [
+    {
+      category: authCategoryName,
+      resourceName: authResourceName,
+      attributes: ['UserPoolId']
+    }
+  ];
+  if (groupNames.length > 0) {
+    dependsOnResources.push({
+      category: authCategoryName,
+      resourceName: 'userPoolGroups',
+      attributes: groupNames.map(group => `${group}GroupRole`)
+    });
+  }
+  return dependsOnResources;
+}
+
+/**
+ * Get the Geo resources added to the project
+ */
+export const getGeoResources = async (service: ServiceName): Promise<string[]> => {
+  const serviceMeta = await getGeoServiceMeta(service);
+  return serviceMeta ? Object.keys(serviceMeta) : [];
+}
+
+
+export const getCLIInputsFilePath = (): string => {
+  return path.join(pathManager.getBackendDirPath(), category, cliInputsFileName);
+};
+
+export const readParamsFromCLIInputs = async (): Promise<$TSObject> => {
+  return JSONUtilities.readJson(getCLIInputsFilePath(), { throwIfNotExist: false }) || {};
+};
+
+export const writeParamsToCLIInputs = async (params: $TSObject, resourceName: string) => {
+  const currentParameters = await readParamsFromCLIInputs();
+  currentParameters[resourceName] = { ...currentParameters[resourceName], ...params };
+  JSONUtilities.writeJson(getCLIInputsFilePath(), currentParameters);
+}
```

Comment:
Use stateManager.getResourceInputsJson() and stateManager.setResourceInputsJson() for getting and setting the contents of `cli-inputs.json`. Also note, there is a `cli-inputs.json` file for each resource within a category, not one per category so the top level resourceName key should be removed here